radixdlt-java/src/main/java/org/radix/serialization2/mapper/RadixCBORGenerator.java

```diff
@@ -0,0 +1,1677 @@
+            int room = _outputEnd - _outputTail;
+            if (room <= 0) {
+                _flushBuffer();
+                room = _outputEnd - _outputTail;
+            }
+            int count = in.read(_outputBuffer, _outputTail, room);
+            if (count < 0) {
+                break;
+            }
+            _outputTail += count;
+            bytesLeft -= count;
+        }
+        return bytesLeft;
+    }
+
+    private final void _writeBytesLong(byte[] data, int offset, int len)
+            throws IOException {
+        if (_outputTail >= _outputEnd) {
+            _flushBuffer();
+        }
+        while (true) {
+            int currLen = Math.min(len, (_outputEnd - _outputTail));
+            System.arraycopy(data, offset, _outputBuffer, _outputTail, currLen);
+            _outputTail += currLen;
+            if ((len -= currLen) == 0) {
+                break;
+            }
+            offset += currLen;
+            _flushBuffer();
+        }
+    }
+
+    /*
+    /**********************************************************
+    /* Internal methods, buffer handling
+    /**********************************************************
+     */
+
+    @Override
+    protected void _releaseBuffers() {
+        byte[] buf = _outputBuffer;
+        if (buf != null && _bufferRecyclable) {
+            _outputBuffer = null;
+            _ioContext.releaseWriteEncodingBuffer(buf);
+        }
+        char[] cbuf = _charBuffer;
+        if (cbuf != null) {
+            _charBuffer = null;
+            _ioContext.releaseConcatBuffer(cbuf);
+        }
+    }
+
+    protected final void _flushBuffer() throws IOException {
+        if (_outputTail > 0) {
+            _bytesWritten += _outputTail;
+            _out.write(_outputBuffer, 0, _outputTail);
+            _outputTail = 0;
+        }
+    }
+
+    /*
+    /**********************************************************
+    /* Internal methods, size control for array and objects
+    /**********************************************************
+	*/
+
+    private final void closeComplexElement() throws IOException {
+        switch (_currentRemainingElements) {
+        case INDEFINITE_LENGTH:
+            _writeByte(BYTE_BREAK);
+            break;
+        case 0: // expected for sized ones
+            break;
+        default:
+            _reportError(String.format("%s size mismatch: expected %d more elements",
+                    _writeContext.typeDesc(), _currentRemainingElements));
+        }
+        _currentRemainingElements = (_elementCountsPtr == 0)
+                ? INDEFINITE_LENGTH
+                        : _elementCounts[--_elementCountsPtr];
+    }
+
+    /*
+    /**********************************************************
+    /* Internal methods, error reporting
+    /**********************************************************
+     */
+
+    protected UnsupportedOperationException _notSupported() {
+        return new UnsupportedOperationException();
+    }
+}
```

Comment:
Holy immaculate mother of God! This was a biggie :)

Reply:
Mostly Jackson code.  As noted in the header comment, our contribution is the 5-line `writeLongString(...)` method :)

Reply:
Yes, this was a superfluous comment from my side. I realise it wasn't our contribution that caused the file to be this big, I was just so surprised (by it's original size) I lost control over my fingers for a moment there :)